server/sonar-server/src/main/java/org/sonar/server/user/index/UserIndexer.java

```diff
@@ -19,35 +19,49 @@
  */
 package org.sonar.server.user.index;
 
+import com.google.common.collect.ImmutableSet;
 import java.util.Iterator;
+import java.util.Set;
 import org.elasticsearch.action.index.IndexRequest;
 import org.sonar.api.utils.System2;
 import org.sonar.db.DbClient;
 import org.sonar.db.DbSession;
-import org.sonar.server.es.BaseIndexer;
 import org.sonar.server.es.BulkIndexer;
 import org.sonar.server.es.EsClient;
+import org.sonar.server.es.IndexTypeId;
+import org.sonar.server.es.StartupIndexer;
 
-public class UserIndexer extends BaseIndexer {
+import static org.sonar.server.user.index.UserIndexDefinition.INDEX_TYPE_USER;
+
+public class UserIndexer implements StartupIndexer {
 
   private final DbClient dbClient;
+  private final EsClient esClient;
 
   public UserIndexer(System2 system2, DbClient dbClient, EsClient esClient) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "system2". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1172)